xbmc/network/httprequesthandler/HTTPImageHandler.cpp

```diff
@@ -46,6 +53,15 @@ CHTTPImageHandler::CHTTPImageHandler(const HTTPRequest &request)
   SetFile(file, responseStatus);
 }
 
+bool CHTTPImageHandler::GetLastModifiedDate(CDateTime &lastModified) const
```

Comment:
The base class `CHTTPFileHandler` already offers helper methods to handle this. You just need to call `SetCanBeCached()` in the constructor and `SetLastModifiedDate()` instead of using your own private `m_lastModified` variable.